In [1]:
ls

sample_data/


In [2]:
!git clone "https://github.com/Shreeyash-iitr/Multimodal-MOSEI.git"

Cloning into 'Multimodal-MOSEI'...
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 16 (delta 5), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (16/16), done.


In [3]:
cd Multimodal-MOSEI/

/content/Multimodal-MOSEI


In [4]:
!git pull origin master

From https://github.com/Shreeyash-iitr/Multimodal-MOSEI
 * branch            master     -> FETCH_HEAD
Already up to date.


In [5]:
cd data/

/content/Multimodal-MOSEI/data


In [0]:
!chmod +x mosei.sh

In [0]:
!./mosei.sh

In [8]:
cd ..

/content/Multimodal-MOSEI


In [15]:
ls data/

audio_test.h5   ey_train.h5       text_train_emb.h5  video_valid.h5
audio_train.h5  ey_valid.h5       text_valid_emb.h5  y_test.h5
audio_valid.h5  mosei.sh*         video_test.h5      y_train.h5
ey_test.h5      text_test_emb.h5  video_train.h5     y_valid.h5


In [0]:
import h5py
import numpy as np
import pandas as pd

In [165]:
y_test = h5py.File("data/y_test.h5","r")
print(list(y_test.keys()))
y_test = np.array(y_test.get('d1'))
df = pd.DataFrame(y_test,columns=["y_test"])
print("\nBEFORE PREPROCESING")
print((df>3).sum())
print(df[df["y_test"]>3])
df[df["y_test"]>3] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()

['d1']

BEFORE PREPROCESING
y_test    1
dtype: int64
        y_test
2747  3.666667

AFTER VALUE CLIPPING


y_test
count  4832.000000
mean      0.126587
std       1.108877
min      -3.000000
25%      -0.333333
50%       0.000000
75%       0.666667
max       3.000000

 As for the sentiments, they should be in range [-3, +3], and the intervals are just evenly divided. We are aware that a few data points seem to have out-of-bounds annotations, and we'll clip them. So you can also just clip those values beyond [-3, 3].


In [166]:
y_train = h5py.File("data/y_train.h5","r")
print(list(y_train.keys()))
y_train = np.array(y_train.get('d1'))
df = pd.DataFrame(y_train,columns=["y_train"])
print("\nBEFORE PREPROCESING")
print((df>3).sum())
print(df[df["y_train"]>3])
df[df["y_train"]>3] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()

['d1']

BEFORE PREPROCESING
y_train    3
dtype: int64
       y_train
5593  3.333333
9663  3.666667
9685  5.000000

AFTER VALUE CLIPPING


y_train
count  15290.000000
mean       0.142991
std        1.123811
min       -3.000000
25%       -0.333333
50%        0.000000
75%        0.666667
max        3.000000

Each sentence is annotated for sentiment on a [-3,3]
Likert scale of: [−3: highly negative, −2 negative,
−1 weakly negative, 0 neutral, +1 weakly positive,
+2 positive, +3 highly positive].

In [167]:
y_valid = h5py.File("data/y_valid.h5","r")
print(list(y_valid.keys()))
y_valid = np.array(y_valid.get('d1'))
df = pd.DataFrame(y_valid,columns=["y_valid"])
print("\nBEFORE PREPROCESING")
print((df>3).sum())
print(df[df["y_valid"]>3])
df[df["y_valid"]>3] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()

['d1']

BEFORE PREPROCESING
y_valid    0
dtype: int64
Empty DataFrame
Columns: [y_valid]
Index: []

AFTER VALUE CLIPPING


y_valid
count  2291.000000
mean      0.219991
std       1.109050
min      -3.000000
25%      -0.333333
50%       0.333333
75%       1.000000
max       3.000000

In [168]:
ey_train = h5py.File("data/ey_train.h5","r")
print(list(ey_train.keys()))
ey_train = np.array(ey_train.get('d1'))
ey_train.shape
df = pd.DataFrame(ey_train,columns=["anger", "disgust", "fear", "happy", "sad", "surprise"])
print("\nBEFORE PREPROCESSING")
print(df.astype(bool).sum(axis=0))
print((df>3).sum())
df.loc[df["happy"]>3,"happy"] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()


['d1']

BEFORE PREPROCESSING
anger       3443
disgust     2720
fear        1319
happy       8147
sad         3906
surprise    1562
dtype: int64
anger       0
disgust     0
fear        0
happy       1
sad         0
surprise    0
dtype: int64

AFTER VALUE CLIPPING


anger       disgust  ...           sad      surprise
count  15290.000000  15290.000000  ...  15290.000000  15290.000000
mean       0.165402      0.122913  ...      0.155853      0.049684
std        0.375969      0.327749  ...      0.322669      0.172264
min        0.000000      0.000000  ...      0.000000      0.000000
25%        0.000000      0.000000  ...      0.000000      0.000000
50%        0.000000      0.000000  ...      0.000000      0.000000
75%        0.000000      0.000000  ...      0.333333      0.000000
max        3.000000      3.000000  ...      3.000000      3.000000

[8 rows x 6 columns]

the emotions for each of the dimensions are: anger, disgust, fear, happy, sad, surprise.


In [172]:
ey_test = h5py.File("data/ey_test.h5","r")
print(list(ey_test.keys()))
ey_test = np.array(ey_test.get('d1'))
print(ey_test.shape)
df = pd.DataFrame(ey_test,columns=["anger", "disgust", "fear", "happy", "sad", "surprise"])
print("\nBEFORE PREPROCESSING")
print(df.astype(bool).sum(axis=0))
print((df>3).sum())
df.loc[df["happy"]>3,"happy"] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()

['d1']
(4832, 6)

BEFORE PREPROCESSING
anger        971
disgust      922
fear         332
happy       2522
sad         1334
surprise     479
dtype: int64
anger       0
disgust     0
fear        0
happy       1
sad         0
surprise    0
dtype: int64

AFTER VALUE CLIPPING


anger      disgust  ...          sad     surprise
count  4832.000000  4832.000000  ...  4832.000000  4832.000000
mean      0.142453     0.128242  ...     0.165701     0.046427
std       0.338588     0.326952  ...     0.316884     0.162014
min       0.000000     0.000000  ...     0.000000     0.000000
25%       0.000000     0.000000  ...     0.000000     0.000000
50%       0.000000     0.000000  ...     0.000000     0.000000
75%       0.000000     0.000000  ...     0.333333     0.000000
max       2.333333     3.000000  ...     2.666667     2.000000

[8 rows x 6 columns]

 Ekman emotions
(Ekman et al., 1980) of {happiness, sadness, anger,
fear, disgust, surprise} are annotated on a [0,3] Likert scale for presence of emotion x: [0: no evidence
of x, 1: weakly x, 2: x, 3: highly x].

In [170]:
ey_valid = h5py.File("data/ey_valid.h5","r")
print(list(ey_valid.keys()))
ey_valid = np.array(ey_valid.get('d1'))
df = pd.DataFrame(ey_valid,columns=["anger", "disgust", "fear", "happy", "sad", "surprise"])
print("\nBEFORE PREPROCESSING")
print(df.astype(bool).sum(axis=0))
print((df>3).sum())
df.loc[df["happy"]>3,"happy"] = 3
print("\nAFTER VALUE CLIPPING")
df.describe()

['d1']

BEFORE PREPROCESSING
anger        427
disgust      352
fear         186
happy       1313
sad          576
surprise     201
dtype: int64
anger       0
disgust     0
fear        0
happy       0
sad         0
surprise    0
dtype: int64

AFTER VALUE CLIPPING


anger      disgust  ...          sad     surprise
count  2291.000000  2291.000000  ...  2291.000000  2291.000000
mean      0.123381     0.096028  ...     0.160483     0.043067
std       0.306181     0.271380  ...     0.355313     0.169733
min       0.000000     0.000000  ...     0.000000     0.000000
25%       0.000000     0.000000  ...     0.000000     0.000000
50%       0.000000     0.000000  ...     0.000000     0.000000
75%       0.000000     0.000000  ...     0.333333     0.000000
max       2.666667     2.666667  ...     3.000000     2.333333

[8 rows x 6 columns]

In [162]:
video_train = h5py.File("data/video_train.h5","r")
print(list(video_train.keys()))
video_train = np.array(video_train.get('d1'))
video_train.shape

['d1']


(15290, 20, 35)

20 means the number of words in a particular training sentence (there is 15290 in that file). If I am not mistaken 35 means the FACET dimensions. That means continuous alignment which CMU used words as reference so basically word alignment. They allow each sentence to have 20 words max. 300 is the glove dimensions. 35 is for the video and 74 is for the audio.

In [137]:
video_test = h5py.File("data/video_test.h5","r")
print(list(video_test.keys()))
video_test = np.array(video_test.get('d1'))
video_test.shape

['d1']


(4832, 20, 35)

In [36]:
video_valid = h5py.File("data/video_valid.h5","r")
print(list(video_valid.keys()))
video_valid = np.array(video_valid.get('d1'))
video_valid.shape

['d1']


(2291, 20, 35)

In [37]:
audio_train = h5py.File("data/audio_train.h5","r")
print(list(audio_train.keys()))
audio_train = np.array(audio_train.get('d1'))
audio_train.shape

['d1']


(15290, 20, 74)

In [38]:
audio_test = h5py.File("data/audio_test.h5","r")
print(list(audio_test.keys()))
audio_test = np.array(audio_test.get('d1'))
audio_test.shape

['d1']


(4832, 20, 74)

In [39]:
audio_valid = h5py.File("data/audio_valid.h5","r")
print(list(audio_valid.keys()))
audio_valid = np.array(audio_valid.get('d1'))
audio_valid.shape

['d1']


(2291, 20, 74)

In [40]:
text_train_emb = h5py.File("data/text_train_emb.h5","r")
print(list(text_train_emb.keys()))
text_train_emb = np.array(text_train_emb.get('d1'))
text_train_emb.shape

['d1']


(15290, 20, 300)

In [41]:
text_test_emb = h5py.File("data/text_test_emb.h5","r")
print(list(text_test_emb.keys()))
text_test_emb = np.array(text_test_emb.get('d1'))
text_test_emb.shape

['d1']


(4832, 20, 300)

In [42]:
text_valid_emb = h5py.File("data/text_valid_emb.h5","r")
print(list(text_valid_emb.keys()))
text_valid_emb = np.array(text_valid_emb.get('d1'))
text_valid_emb.shape

['d1']


(2291, 20, 300)